src/main/kotlin/org/rust/cargo/toolchain/Rustfmt.kt

```diff
@@ -27,7 +28,11 @@ class Rustfmt(private val rustfmtExecutable: Path) {
         if (file.isNotRustFile || !file.isValid) return null
 
         val configPath = findConfigPathRecursively(file.parent, stopAt = cargoProject.workingDirectory)
-        return reformatText(cargoProject, document.text, configPath)
+
+        val psiFile = file.toPsiFile(cargoProject.project)
+        val isEdition2018 = psiFile?.isEdition2018 ?: true
```

Comment:
You het exception after `Reformat File with Rustfmt` action invocation:
```
java.lang.Throwable: Read access is allowed from event dispatch thread or inside read-action only (see com.intellij.openapi.application.Application.runReadAction())
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:162)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertReadAccessAllowed(ApplicationImpl.java:1027)
	at com.intellij.psi.impl.file.impl.FileManagerImpl.findFile(FileManagerImpl.java:364)
	at com.intellij.psi.impl.PsiManagerImpl.findFile(PsiManagerImpl.java:160)
	at org.rust.openapiext.UtilsKt.toPsiFile(utils.kt:118)
	at org.rust.cargo.toolchain.Rustfmt.reformatDocumentText(Rustfmt.kt:32)
	at org.rust.ide.actions.RustfmtFileAction.reformatDocumentAndGetText(RustfmtFileAction.kt:48)
	at org.rust.ide.actions.RustfmtFileAction.access$reformatDocumentAndGetText(RustfmtFileAction.kt:29)
	at org.rust.ide.actions.RustfmtFileAction$actionPerformed$formattedText$1.invoke(RustfmtFileAction.kt:40)
	at org.rust.ide.actions.RustfmtFileAction$actionPerformed$formattedText$1.invoke(RustfmtFileAction.kt:29)
	at org.rust.openapiext.UtilsKt$sam$com_intellij_openapi_util_ThrowableComputable$0.compute(utils.kt)
	at com.intellij.openapi.progress.impl.CoreProgressManager$1.run(CoreProgressManager.java:225)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:894)
	at com.intellij.openapi.progress.impl.CoreProgressManager$5.run(CoreProgressManager.java:447)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:169)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:156)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$null$9(ApplicationImpl.java:552)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:294)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Consider wrapping this code into read action. But check that it's ok to get read action here in all cases